modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotDeliveryWorker.java

```diff
@@ -87,6 +88,13 @@ public void run() {
                         if (messageFlusher.getMessageDeliveryInfo(destinationOfMessagesInQueue)
                                 .isMessageBufferFull()) {
 
+                            //If hazelcast is inactive at this moment, we cannot proceed further with the clustered
+                            // data collections. therefore warning and skipping.
+                            if (AndesContext.getInstance().isClusteringEnabled() && !HazelcastAgent.getInstance().isActive()) {
```

Comment:
It not good to check it here like that. Because hazelcast instance can go down just after this line and therefore exception can occur. Better to use a try catch and catch the HazelcastInstanceNotActiveException and ignore the exception is the server is shutting down. Otherwise print a error log.